evalml/utils/woodwork_utils.py

```diff
@@ -66,7 +66,7 @@ def infer_feature_types(data, feature_types=None):
 
     def convert_all_nan_unknown_to_double(data):
         def is_column_pd_na(data, col):
-            return all([isinstance(x, type(pd.NA)) for x in data[col]])
+            return all(data[col].isna())
```

Comment:
@freddyaboulton and @bchen1116 I think you recommended both this and `all(data[col] == pd.NA)`.  What happened was I tried `all(data[col] == pd.NA)`, which created weird behavior in as much as a column full of `pd.NA` returns a column full of `pd.NA` when compared to `pd.NA`.  I found a little bit more about this [here](https://github.com/pandas-dev/pandas/issues/38063), where they discuss some of the design decisions about how arithmetic and comparisons using `pd.NA` result in `pd.NA` to "propagate pd.NA".  I ended up just going with what I had and it was super slow - as you foresaw.  I'm now trying to atone for my sins.

Reply:
Great catch.